File: src/main/java/net/lumalyte/hooks/NexoHook.java
package net.lumalyte.hooks;

import com.nexomc.nexo.api.NexoItems;
import net.lumalyte.LumaSG;
import net.lumalyte.util.DebugLogger;
import org.bukkit.Bukkit;
import org.bukkit.inventory.ItemStack;
import org.bukkit.plugin.Plugin;

import java.util.Objects;
import java.util.Optional;

/**
 * Hook for the Nexo plugin.
 */
public class NexoHook implements PluginHook {
    
    private static final String PLUGIN_NAME = "Nexo";
    private Plugin nexoPlugin;
    private boolean available = false;
    
    /** The debug logger instance for this nexo hook */
    private final DebugLogger.ContextualLogger logger;
    
    public NexoHook(LumaSG plugin) {
        this.logger = plugin.getDebugLogger().forContext("NexoHook");
    }
    
    @Override
    public String getPluginName() {
        return PLUGIN_NAME;
    }
    
    @Override
    public boolean isAvailable() {
        return available;
    }
    
    @Override
    public Plugin getPlugin() {
        return nexoPlugin;
    }
    
    @Override
    public boolean initialize() {
        nexoPlugin = Bukkit.getPluginManager().getPlugin(PLUGIN_NAME);
        
        if (nexoPlugin == null || !nexoPlugin.isEnabled()) {
            logger.info("Nexo plugin not found or not enabled. Nexo item support will be disabled.");
            return false;
        }
        
        // Test if we can access the Nexo API by attempting to use it
        try {
            // Try to get a non-existent item - this will throw a normal ItemNotFoundException
            // but will validate that we can access the API
            NexoItems.itemFromId("test_api_access");
            available = true;
            logger.info("Successfully hooked into Nexo!");
            return true;
        } catch (IllegalArgumentException e) {
            // Expected exception for non-existent item - API is working
            available = true;
            logger.info("Successfully hooked into Nexo!");
            return true;
        } catch (Exception e) {
            // Any other exception means we can't access the API
            logger.warn("Failed to access Nexo API. Nexo item support will be disabled.", e);
            return false;
        }
    }
    
    /**
     * Gets a Nexo item by its ID.
     * 
     * @param itemId The ID of the Nexo item
     * @return The ItemStack, or empty if the item could not be found
     */
    public Optional<ItemStack> getNexoItem(String itemId) {
        if (!isAvailable()) {
            return Optional.empty();
        }
        
        try {
            return Optional.of(Objects.requireNonNull(NexoItems.itemFromId(itemId)).build());
        } catch (Exception e) {
            logger.warn("Failed to get Nexo item: " + itemId, e);
            return Optional.empty();
        }
    }

    @Override
    public boolean enable() {
        return initialize();
    }

    @Override
    public void disable() {
        available = false;
        nexoPlugin = null;
    }
} 

File: src/main/kotlin/net/lumalyte/lumasg/hooks/NexoHook.kt
package net.lumalyte.lumasg.hooks

import com.nexomc.nexo.api.NexoItems
import net.badgersmc.nexus.annotations.PostConstruct
import net.badgersmc.nexus.annotations.Service
import org.bukkit.inventory.ItemStack
import org.bukkit.plugin.java.JavaPlugin

@Service
class NexoHook(private val plugin: JavaPlugin) : PluginHook {
    override val pluginName = "Nexo"

    private var nexoAvailable = false

    @PostConstruct
    fun register() {
        if (!isAvailable()) {
            plugin.logger.info("Nexo not found — custom item compatibility disabled.")
            return
        }
        nexoAvailable = true
        plugin.logger.info("Nexo integration enabled.")
    }

    /**
     * Get a Nexo item by its ID.
     * Returns null if Nexo is not available or the item doesn't exist.
     */
    fun getNexoItem(id: String): ItemStack? {
        if (!nexoAvailable) return null
        return try {
            NexoItems.itemFromId(id)?.build()
        } catch (_: Exception) {
            null
        }
    }

    /** Check whether a Nexo item ID exists. */
    fun exists(id: String): Boolean {
        if (!nexoAvailable) return false
        return try {
            NexoItems.exists(id)
        } catch (_: Exception) {
            false
        }
    }
}
